plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/handlers/SQLEditorHandlerExecute.java

```diff
@@ -52,6 +59,38 @@ public Object execute(ExecutionEvent event) throws ExecutionException
             case SQLEditorCommands.CMD_EXECUTE_SCRIPT:
                 editor.processSQL(false, true);
                 break;
+            case SQLEditorCommands.CMD_EXECUTE_SCRIPT_NATIVE:
+                SQLNativeExecutorDescriptor executorDescriptor = SQLNativeExecutorRegistry.getInstance()
+                    .getExecutorDescriptor(editor.getDataSource());
+                if (executorDescriptor == null) {
+                    throw new ExecutionException("Valid native executor not found");
+                }
+                try {
+                    if (editor.getExecutionContext() instanceof DBCExecutionContextDefaults) {
+                        DBCExecutionContextDefaults<?, ?> executionContext
+                            = (DBCExecutionContextDefaults<?, ?>) editor.getExecutionContext();
+                        if (executionContext.getDefaultCatalog() != null) {
+                            SQLEditorExecutor<DBSCatalog> nativeExecutor
+                                = (SQLEditorExecutor<DBSCatalog>) executorDescriptor.getNativeExecutor();
+                            if (nativeExecutor == null) {
+                                throw new ExecutionException("Valid native executor not found");
+                            }
+
+                            nativeExecutor.execute(executionContext.getDefaultCatalog(), editor);
+                        } else {
+                            SQLEditorExecutor<DBPDataSource> nativeExecutor =
+                                (SQLEditorExecutor<DBPDataSource>) executorDescriptor.getNativeExecutor();
+                            if (nativeExecutor == null) {
+                                throw new ExecutionException("Valid native executor not found");
+                            }
+                            nativeExecutor.execute(editor.getDataSource(), editor);
+                        }
```

Comment:
What's going on.

You DON'T need to specify the generic argument to be something specific as `DBSCatalog` or `DBPDataSource`. Just cast it to `Object` and pass whatever you want as an argument.